File: server/Core/Logger.php
<?php

// Уровни логирования, минимальный уровень берётся из config.php ('log_level').
// По умолчанию — debug: пока приложение в бете, лучше больше сигнала в логах, чем
// меньше, когда баги ещё ловятся на реальных кассах, а не в проде.
class Logger
{
    const MAX_BYTES = 20971520; // 20 МБ

    private static $levels = array(
        'debug'   => 10,
        'info'    => 20,
        'warning' => 30,
        'error'   => 40,
    );

    private static $minLevel = null;

    public static function debug($message)
    {
        self::write('debug', $message);
    }

    public static function info($message)
    {
        self::write('info', $message);
    }

    public static function warning($message)
    {
        self::write('warning', $message);
    }

    public static function error($message)
    {
        self::write('error', $message);
    }

    private static function write($level, $message)
    {
        if (self::$levels[$level] < self::minLevel()) {
            return;
        }

        $line = '[' . date('Y-m-d H:i:s') . '] [' . strtoupper($level) . '] ' . $message . PHP_EOL;
        $path = __DIR__ . '/../logs/app.log';

        // Ротация по размеру: на 3000 касс debug-лог растёт на десятки мегабайт в час,
        // без ротации диск сервера кончится за пару недель. Храним текущий + один .old.
        if (file_exists($path) && filesize($path) > self::MAX_BYTES) {
            @rename($path, $path . '.old');
        }

        file_put_contents($path, $line, FILE_APPEND | LOCK_EX);
    }

    private static function minLevel()
    {
        if (self::$minLevel === null) {
            $configured = strtolower((string) Config::get('log_level'));
            self::$minLevel = isset(self::$levels[$configured]) ? self::$levels[$configured] : self::$levels['debug'];
        }

        return self::$minLevel;
    }
}

File: client-dotnet/Core/Logger.cs
using System;
using System.IO;

namespace AMadmin.Core
{
    public enum LogLevel { Debug = 10, Info = 20, Warning = 30, Error = 40 }

    // Лог в файл с уровнями и простой ротацией (один .old при превышении размера) — тот же
    // принцип, что и на сервере. По умолчанию Debug, пока приложение в бете.
    public static class Logger
    {
        private const long MaxBytes = 1024 * 1024;
        private static readonly object Sync = new object();

        public static string Path { get; set; } = "agent.log";
        public static LogLevel MinLevel { get; set; } = LogLevel.Debug;

        public static void SetLevel(string level)
        {
            LogLevel parsed;
            MinLevel = Enum.TryParse(level, true, out parsed) ? parsed : LogLevel.Debug;
        }

        public static void Debug(string message) { Write(LogLevel.Debug, message); }
        public static void Info(string message) { Write(LogLevel.Info, message); }
        public static void Warning(string message) { Write(LogLevel.Warning, message); }
        public static void Error(string message) { Write(LogLevel.Error, message); }

        private static void Write(LogLevel level, string message)
        {
            if (level < MinLevel) return;

            var line = "[" + DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss") + "] [" +
                       level.ToString().ToUpperInvariant() + "] " + message + Environment.NewLine;

            lock (Sync)
            {
                try
                {
                    var info = new FileInfo(Path);
                    if (info.Exists && info.Length > MaxBytes)
                    {
                        File.Copy(Path, Path + ".old", true);
                        File.Delete(Path);
                    }
                    File.AppendAllText(Path, line);
                }
                catch
                {
                    // Логирование не должно ронять агента — если диск недоступен, просто молчим.
                }
            }
        }
    }
}
